terraform/cloud-platform-components/README.md

```diff
@@ -82,6 +83,77 @@ Metrics-server allows us to perform resource queries against the cluster. Comman
+
+In Kubernetes, Admission Controllers enforce semantic validation of objects during create, update, and delete operations. With OPA we can enforce custom policies on Kubernetes objects without recompiling or reconfiguring the Kubernetes API server. Please see the following documentation by opa on Kubernetes Admission Control 
+https://www.openpolicyagent.org/docs/kubernetes-admission-control.html
+
+To restrict the kinds of operations and resources that are subject to OPA policy checks, see the below example. Configured in templates/opa/values.yaml.tpl under admissionControllerRules:
+
+```yaml
+    admissionControllerRules:
+      - operations: ["CREATE", "UPDATE"]
+        apiGroups: ["extensions", ""]
+        apiVersions: ["v1beta1”, “v1”]
+        resources: ["ingresses", "namespaces"]
+```
+The admissionControllerRules defines the operations and resources that the webhook will validate. Intercept API requests when a ingress or a namespace is CREATED or UPDATED, so apiGroups and apiVersions are filled out accordingly (extensions/v1beta1 for ingresses, v1 for namespaces). We can use wildcards (*) for these fields as well.
+
+ ### kube-mgmt
+
+kube-mgmt manages instances of the Open Policy Agent on top of kubernetes. Use kube-mgmt to:
+
+  - Load policies into OPA via kubernetes.
+  - Replicate kubernetes resources including CustomResourceDefinitions (CRDs) into OPA.
+
+kube-mgmt scans are configured in templates/opa/values.yaml.tpl under mgmt/replicate:
+
+  ```yaml
+    group/version/resource			  # namespace scoped
+    group/version/resource        # cluster scoped
+  ```
+    <replicate-path> is configurable (via path) and defaults to kubernetes.
+
+The example below would replicate namespaces and ingress into OPA:
+
+```yaml
+    cluster:
+      - "v1/namespaces"
+    namespace:
+      - "extensions/v1beta1/ingresses"
+    path: kubernetes
+```
+
+NOTE: IF we use replicate: remember to update the RBAC rules to allow permissions to replicate these things
+
+ ### opa-policies
+
+kube-mgmt automatically discovers policies stored in ConfigMaps in kubernetes and loads them into OPA. kube-mgmt assumes a ConfigMap contains policies if the ConfigMap is :
+
+  - Created in a namespace listed in the --policies option. Configured in templates/opa/values.yaml.tpl under mgmt/configmapPolicies/namespaces:
+	- Labelled with openpolicyagent.org/policy=rego.
+
+
+When a policy has been successfully loaded into OPA, the openpolicyagent.org/policy-status annotation is set to
+```json
+{"status": "ok"}.
+```
+If loading fails for some reason (e.g., because of a parse error), the openpolicyagent.org/policy-status annotation is set to 
+```json
+{"status": "error", "error": ...} where the error field contains details about the failure.
```

Comment:
Can you please move the sentence outside of the code block?

Reply:
moved the sentence outside of the code block